---
title: "Cancer Genomics Tutorial"
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    dev: 'svg'
  md_document:
    variant: gfm
date: '2022-04-25'
---

SECTION 6: Reading and Summarizing MAF Files

The input files for reading/summarizing MAF files are a MAF file (can be gz compressed), an optional clinical data associated with each sample, and an optional copy number data file. In this section, we read the MAF file and viewed the sample summary, the gene summary, the clinical data, and the fields. 

```{r}
# 6.2: Reading MAF files
library(maftools)
clinical.data = read.table("tcga_laml_annot.tsv",sep
                           ="\t", header = TRUE) #read in metadata
laml = read.maf(maf = "tcga_laml.maf", clinicalData = clinical.data) #create a MAF object
relapse.apl = read.maf(maf = "APL_primary.maf")
APL_relapse.maf = read.maf(maf = "APL_relapse.maf")
brca = read.maf(maf = "brca.maf")

# 6.3: MAF Object
laml
getSampleSummary(laml)
getGeneSummary(laml)
getClinicalData(laml)
getFields(laml)
write.mafSummary(maf = laml, basename = 'laml')
```

SECTION 7: Visualization

In this section, we visualized the MAF data in various different ways. In 7.1, we first plotted the MAF summary: the number of variants in each sample is represented as a stacked bar plot and the variant types are represented by the box plot. Additionally, in 7.2, we drew oncoplots, or waterfall plots, which are a better representation of a maf file. In this oncoplot, we can see the frequency of mutations in the genes. We then classified SNPs into transitions and transversions in 7.3 using the titv function and ploted the summarized data as a boxplot showing overall distribution of six different conversions and as a stacked barplot showing fraction of conversions in each sample. 

In 7.4, we created a lollipop plot in order to visualize the amino acid changes. This can be challenging because MAF files have no clear guidelines on naming the amino acid change field: the lolliplot function by default looks for the AAChange column, and if that is not found, it prints all available fields with a warning message. We then drew the general protein domains using plotProtein. In 7.5, we created rainfall plots. Because cancer genomes (especially solid tumors) are typically characterized by localized hyper-mutations, rainfall plots are especially useful. Rainfall plots can visualize hypermutated genomic regions by plotting inter-variant distance on a linear genomic scale. In section 7.6, we compared the mutation load against TCGA cohorts using tcgaCompare and plotted the results. Finally, we plotted the variant allele frequences (VAF) as a boxplot in order to estimate the clonal status of top mutated genes. Clonal genes typically have a mean allele frequency of around 50% (assuming pure samples).

```{r}
# 7.1 Plotting MAF Summary
plotmafSummary(maf = laml, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
# 7.2: Drawing oncoplots
oncoplot(maf = laml, top = 10)
# 7.3: Transition and Transversions
laml.titv = titv(maf = laml, plot = FALSE, useSyn = TRUE)
plotTiTv(res = laml.titv)
# 7.4: Lolliplot plots for amino acid changes
lollipopPlot(
  maf = laml,
  gene = 'DNMT3A',
  AACol = 'Protein_Change',
  showMutationRate = TRUE,
  labelPos = 882
)
plotProtein(gene = "TP53", refSeqID = "NM_000546")
# 7.5: Rainfall Plots
rainfallPlot(maf = brca, detectChangePoints = TRUE, pointSize = 0.4)
# 7.6 Compare mutation load against TCGA cohorts
laml.mutload = tcgaCompare(maf = laml, cohortName = 'Example-LAML', logscale = TRUE, capture_size = 50)
# Plotting VAF
graphics.off()
par(mar=c(1,1,1,1))
plotVaf(maf = laml, vafCol = 'i_TumorVAF_WU')
```

SECTION 8

In this section, we processed the copy-number data and visualized it through various plots. First, in 8.1, we read and summarized the GISTIC output files. First, we needed to obtain the four files that were generated by GISTIC and view the GISTIC object: all_lesions.conf_XX.txt, amp_genes.conf_XX.txt, del_genes.conf_XX.txt and scores.gistic. We then visualized the gistic results using genome plot (8.2.1), bubble plot (8.2.2), and oncoplot (8.2.3). We can also visualize the CBS segments (8.2.4).

```{r}
# 8.1: Reading and summarizing gistic output files
all.lesions <- system.file("extdata", "all_lesions.conf_99.txt", package = "maftools")
amp.genes <- system.file("extdata", "amp_genes.conf_99.txt", package = "maftools")
del.genes <- system.file("extdata", "del_genes.conf_99.txt", package = "maftools")
scores.gis <- system.file("extdata", "scores.gistic", package = "maftools")

laml.gistic = readGistic(gisticAllLesionsFile = all.lesions, gisticAmpGenesFile = amp.genes, gisticDelGenesFile = del.genes, gisticScoresFile = scores.gis, isTCGA = TRUE)

laml.gistic
# 8.2: Visualizing gistic results
# 8.2.1: Genome plot 
gisticChromPlot(gistic = laml.gistic, markBands = "all")
# 8.2.2: Bubble plot
gisticBubblePlot(gistic = laml.gistic)
#8.2.3: Oncoplot
gisticOncoPlot(gistic = laml.gistic, clinicalData = getClinicalData(x = laml), clinicalFeatures = 'FAB_classification', sortByAnnotation = TRUE, top = 10)

```

SECTION 9: Analysis

In section 9.1, we observed the somatic interactions  between genes: we detected the mutually exclusive/co-occuring sets of genes by utilizing the somaticInteractions function. This function performs pair-wise Fisher's Exact test to detect significant pairs of genes. In section 9.2, we detected cancer driver genes based on positional clustering through the oncodrive function and plotted the results. In section 9.3, we added and summarized pfam domain information to the amino acid changes. This function also summarizes amino acid changes according to the domains that were affected in order to determine what domain in cancer is most frequently affected.  In section 9.4, we performed survival analysis and drew Kaplan Meier curves by grouping samples based on mutation status of genes. First, in 9.4.1, we identified the number of mutated samples for the given genes and plotted the survival probability. Then, in 9.4.2, we predicted the genesets associated with survival. In section 9.5, we compared two cohorts to detect differentially mutated genes between patients who have initial APL and relapsed APL. In 9.5.1, we visualized the difference between two genes, PML and RARA, between Primary APL and Relapse APL patient using a forest plot. In 9.5.2, we utilized an alternative method of displaying these results by plotting two oncoplots side by side. Additionally, we created co-bar plots of the genes in 9.5.3 and lollipop plots in 9.5.4 to show gene-wise differences. 

In section 9.6, we performed clinical enrichment analysis, which performs various groupwise/pairwise comparisons to identify enriched mutations for every category within a clinical feature (ie. FAB classification). In 9.7, we checked for drug-gene interactions and gene drugability information by pulling information compiled from the Drug Gene Interaction database. In section 9.8, we utilized the OncogenicPathways function to check for enrichment of known Oncogenic Signaling Pathway within TCGA cohorts. 

```{r}
# 9.1: Somatic Interactions
somaticInteractions(maf = laml, top = 25, pvalue = c(0.05, 0.1))
# 9.2: Detecing cancer driver genes based on positional clustering
laml.sig = oncodrive(maf = laml, AACol = 'Protein_Change', minMut = 5, pvalMethod = 'zscore')
head(laml.sig)
plotOncodrive(res = laml.sig, fdrCutOff = 0.1, useFraction = TRUE, labelSize = 0.5)
# 9.3: Adding and summarizing pfam domains
laml.pfam = pfamDomains(maf = laml, AACol = 'Protein_Change', top = 10)
laml.pfam$proteinSummary[,1:7, with = FALSE]
laml.pfam$domainSummary[,1:3, with = FALSE]
# 9.4: Survival analysis
# 9.4.1: Mutation in any given genes
mafSurvival(maf = laml, genes = 'DNMT3A', time = 'days_to_last_followup', Status = 'Overall_Survival_Status', isTCGA = TRUE)
# 9.4.2: Predict genesets associated with survival
prog_geneset = survGroup(maf = laml, top = 20, geneSetSize = 2, time = "days_to_last_followup", Status = "Overall_Survival_Status", verbose = FALSE)
print(prog_geneset)

mafSurvGroup(maf = laml, geneSet = c("DNMT3A", "FLT3"), time = "days_to_last_followup", Status = "Overall_Survival_Status")

# 9.5 Comparing two cohorts (MAFs)
#Primary APL MAF
primary.apl = system.file("extdata", "APL_primary.maf.gz", package = "maftools")
primary.apl = read.maf(maf = primary.apl)
#Relapse APL MAF
relapse.apl = system.file("extdata", "APL_relapse.maf.gz", package = "maftools")
relapse.apl = read.maf(maf = relapse.apl)
#Considering only genes which are mutated in at-least in 5 samples in one of the cohort to avoid bias due to genes mutated in single sample.
pt.vs.rt <- mafCompare(m1 = primary.apl, m2 = relapse.apl, m1Name = 'Primary', m2Name = 'Relapse', minMut = 5)
print(pt.vs.rt)
# 9.5.1: Forest plots
forestPlot(mafCompareRes = pt.vs.rt, pVal = 0.1)
# 9.5.2: Co-onco plots
genes = c("PML", "RARA", "RUNX1", "ARID1B", "FLT3")
coOncoplot(m1 = primary.apl, m2 = relapse.apl, m1Name = 'PrimaryAPL', m2Name = 'RelapseAPL', genes = genes, removeNonMutated = TRUE)
# 9.5.3: Co-bar plots
coBarplot(m1 = primary.apl, m2 = relapse.apl, m1Name = "Primary", m2Name = "Relapse")
# 9.5.4 Lollipop plot-2
lollipopPlot2(m1 = primary.apl, m2 = relapse.apl, gene = "PML", AACol1 = "amino_acid_change", AACol2 = "amino_acid_change", m1_name = "Primary", m2_name = "Relapse")
# 9.6: Clinical enrichment analysis
fab.ce = clinicalEnrichment(maf = laml, clinicalFeature = 'FAB_classification')
fab.ce$groupwise_comparision[p_value < 0.05]
plotEnrichmentResults(enrich_res = fab.ce, pVal = 0.05, geneFontSize = 0.5, annoFontSize = 0.6)
# 9.7: Drug-Gene Interactions
dgi = drugInteractions(maf = laml, fontSize = 0.75)
dnmt3a.dgi = drugInteractions(genes = "DNMT3A", drugs = TRUE)
dnmt3a.dgi[,.(Gene, interaction_types, drug_name, drug_claim_name)]
# 9.8: Oncogenic Signaling Pathways
OncogenicPathways(maf = laml)
PlotOncogenicPathways(maf = laml, pathways = "RTK-RAS")
```


