# Module 11: Cancer Genomics

## Author: Kai Saito

Overview: In this assignment, we completed sections 6,7,8, and part 9 (omit 9.9) of the maftools vingette. 
In this tutorial, we became familiar with maftools and used this package to summarize, annotate, and visualize MAF
files from either TCGA sources or in-house studies. This package can be very useful for downstream analysis of data
from somatic mutations in cancer samples. It can facilitate survival analysis, drive gene detection(based on 
positional clustering), estimation of tumor heterogeneity, and visualization of somatic variants. 

## Requirements to run code:
Microsoft Windows 8, macOS 10.13 or later. maftools must be installed.

## References
David Tamborero, Abel Gonzalez-Perez, Nuria Lopez-Bigas, OncodriveCLUST: exploiting the positional clustering of somatic mutations to identify cancer genes, Bioinformatics, Volume 29, Issue 18, 15 September 2013, Pages 2238–2244, https://doi.org/10.1093/bioinformatics/btt395

Griffith, M., Griffith, O., Coffman, A. et al. DGIdb: mining the druggable genome. Nat Methods 10, 1209–1210 (2013). https://doi.org/10.1038/nmeth.2689

Mayakonda A, Lin DC, Assenov Y, Plass C, Koeffler HP. 2018. Maftools: efficient and comprehensive analysis of somatic variants in cancer. Genome Resarch. PMID: 30341162

